com.unity.netcode.gameobjects/Runtime/NetworkVariable/ClientNetworkVariable.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// A ClientNetworkVariable is special in that:
+    ///  - only the owner of the variable can write to it
+    ///  - not even the server can write to it
+    ///  - it is not snapshotted
+    ///  - it must be sent reliably
+    ///
+    /// (This class may be removed in the future when integrated into NetworkVariable natively)
+    /// </summary>
+    [Serializable]
+    public class ClientNetworkVariable<T> : NetworkVariable<T> where T : unmanaged
+    {
+        /// <summary>
+        /// Creates a ClientNetworkVariable with the default read permission
+        /// </summary>
+        public ClientNetworkVariable()
+        {
+        }
+
+        /// <summary>
+        /// Creates a ClientNetworkVariable with a initial value
+        /// </summary>
+        /// <param name="value">The initial value to use for the ClientNetworkVariable</param>
+        public ClientNetworkVariable(T value) : base(value)
+        {
+        }
+
+        /// <summary>
+        /// Creates a ClientNetworkVariable with a specified read permission
+        /// </summary>
+        /// <param name="readPerm">The readPermission to use</param>
+        public ClientNetworkVariable(NetworkVariableReadPermission readPerm) : base(readPerm)
+        {
+        }
+
+        /// <summary>
+        /// Creates a ClientNetworkVariable with a initial value and the specified read permission
+        /// </summary>
+        /// <param name="readPerm">The initial read permission to use for the ClientNetworkVariable</param>
+        /// <param name="value">The initial value to use for the ClientNetworkVariable</param>
+        public ClientNetworkVariable(NetworkVariableReadPermission readPerm, T value) : base(readPerm, value)
+        {
+        }
+
+        public override bool CanClientWrite(ulong clientId)
+        {
+            return m_NetworkBehaviour.OwnerClientId == clientId;
+        }
+
+        public override bool ShouldWrite(ulong clientId, bool isServer)
+        {
+            return m_IsDirty && !isServer && m_NetworkBehaviour.IsOwner;
```

Comment:
A host wouldn't be able to write? If my character is client driven, I don't want different code for my host and my clients, both should be able to update their own characters.

Reply:
Yeah, hosts should be able to write these, although I still yearn for the day when a "host" is just a client and a server running in the same process but each having their own version of each object.